bin/test-dag-tags.py

```diff
@@ -0,0 +1,52 @@
+#!/bin/python3
+from collections import defaultdict
+import subprocess
+
+
+VALID_TAGS = ("impact/tier_1", "impact/tier_2", "impact/tier_3", "repo/bigquery-etl", "repo/telemetry-airflow",)
+REQUIRED_TAG_TYPES = ("impact",)
+
+
+class DagValidationError(Exception):
+    pass
+
+
+def retrieve_existing_airflow_dags():
+    output = subprocess.run(
+        "docker-compose exec web mysql -Ns -h db -u root -psecret airflow -e 'SELECT dag_id, name FROM dag_tag;'",
```

Comment:
Worth exploring if the airflow command line is able to get you tag information instead.

Reply:
I checked, unfortunately it is not possible.

Reply:
I would parse the python files on the filesystem rather than relying on a mysql table. The table may change over time, and also dag state for deleted dags, etc. is not always clear. Also we have multiple github repos

Reply:
Does the circleCi environment even have the dags loaded into mysql?

Reply:
yes, it happens automatically when airflow spins up. I assume that's actually how airflow command for listing dags works. It most likely fetches them from the `dags` table.

Anyways, I added an alternative approach to checking dag tags.

Reply:
Please let me know what you think and then I'll make final tweaks.

Reply:
added comments. I still prefer the non sql approach. we can let klukas chime in also

Reply:
after some thought, i dont feel too strongly about this. If it breaks we can just repair

Reply:
I prefer the sql approach, as it involves working with parsed DAGs. The textual approach could have a lot of unexpected edge cases and will be brittle based on having to maintain mappings from enum constructs to text values.

Reply:
The sql approach is more correct. The parsing would be quicker, and less brittle imo. I don't feel strongly either way, and will rely on kik to be the go to person for ci issues regarding this in the future

Reply:
🙈 <- literally me right now 😅

Reply:
if you go sql, try catch your subprocess.run and print out error codes, exceptions etc.

Reply:
So in this case I made a small tweak to the `.run` call, I believe there is no need for any additional error handling here for that specific command and output should be redirected to `STDOUT` and exception should be raised if `non-zero` return code is present for the shell command.

Reply:
apologies, it looks like I actually have to manually output the errors. Adding that now.